conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosIndexer.scala

```diff
@@ -28,166 +27,42 @@ import tech.cryptonomic.conseil.indexer.config.{
 }
 import tech.cryptonomic.conseil.indexer.logging.LorreProgressLogging
 import tech.cryptonomic.conseil.indexer.tezos.TezosErrors._
-import tech.cryptonomic.conseil.indexer.tezos.TezosNodeOperator.LazyPages
 
 import scala.annotation.tailrec
-import scala.collection.SortedSet
 import scala.concurrent.duration.{Duration, _}
 import scala.concurrent.{Await, ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
+import tech.cryptonomic.conseil.indexer.tezos.processing.BakingAndEndorsingRightsProcessing
+import tech.cryptonomic.conseil.indexer.tezos.processing.AccountsEventsProcessing
+import tech.cryptonomic.conseil.indexer.tezos.processing.BakersProcessing
+import tech.cryptonomic.conseil.indexer.tezos.processing.AccountsProcessing
+import tech.cryptonomic.conseil.indexer.tezos.processing.BlocksProcessing
 
 /** * Class responsible for indexing data for Tezos BlockChain */
-class TezosIndexer(
+class TezosIndexer private (
```

Comment:
BTW, I am fine with keeping it stateful if you think the functions will become too clumsy otherwise. However, I do think there's a elegance to piping the same state through multiple function compositions, with each function acting on the result of the previous computation. Maybe something like a State monad?

Reply:
I can't help but feel there is a good opportunity to describe the Lorre logic as a computation pipeline and have a single for-expression that captures the essence of that pipeline.

Reply:
I also feel the logic of each Lorre iteration can be captured as a very descriptive for-expression, something like:

```scala
for {
blocks <- fetchBlocks()
ops <- fetchOperations()
accounts <- fetchAccounts()
...
_ <- writeToDatabase(blocks, ops, accounts, ..)
}(yield ???)
```

I realize this is a simple refactor PR so, if my comments make any sense, I recommend only following them it it's not more than a couple of days' effort.

Reply:
Makes sense overall, yet I'd need to think over it first a little more.
At first look I'd say that the indexer is indeed stateless, all it needs are input parameters representing configuration and unavoidable external resource (db, node calls, ...).
I don't know how this fits with your comment, please elaborate more if you will.